src/testdir/test_cmdline.vim

```diff
@@ -2042,16 +2042,52 @@ func Wildmode_tests()
   call assert_equal('AAA    AAAA   AAAAA', g:Sline)
   call assert_equal('"b A', @:)
 
+  " When 'wildmenu' is not set, 'noselect' completes first item
+  set wildmode=noselect
+  call feedkeys(":MyCmd o\t\<C-B>\"\<CR>", 'xt')
+  call assert_equal('"MyCmd oneA', @:)
+
+  " When 'noselect' is present, do not complete first <tab>.
+  set wildmenu
+  set wildmode=noselect
+  call feedkeys(":MyCmd o\t\<C-B>\"\<CR>", 'xt')
+  call assert_equal('"MyCmd o', @:)
+  call feedkeys(":MyCmd o\t\t\<C-B>\"\<CR>", 'xt')
+  call assert_equal('"MyCmd o', @:)
+  call feedkeys(":MyCmd o\t\t\<C-Y>\<C-B>\"\<CR>", 'xt')
+  call assert_equal('"MyCmd o', @:)
+
+  " When 'full' is present, complete after first <tab>.
+  set wildmode=noselect,full
```

Comment:
Seems to be order-sensitive. Does the test pass if this line is changed to:

    set wildmode=full,noselect

Reply:
Isn't this just how `'wildmode'` works?

Reply:
That means `set wildmode+=noselect` doesn't work with the default value. Seems strange, and probably should be documented if this is expected.

Reply:
Doesn't the docs say this? 

> It is a comma-separated list of up to four parts.  Each part specifies what to do for each consecutive use of 'wildchar'.  The first part specifies the behavior for the first use of 'wildchar', The second part for the second use, etc.

It has always been order sensitive and the examples also don't use `+=`. I do wonder if it `noselect` ever makes sense to be not used as the first item in the element.

Reply:
No, not really. The docs you quoted are maybe barely understandable in these terms *if* you already know what you are looking for.

If `set wildmode+=...` produces nonsense it should be an error, instead of pretending to support `+=`.

Reply:
Sure, I'm just pointing out that it's not specific to this `noselect` change, as the other values there are already order dependent for this option. That part of the docs has always been a bit hard to parse. I tried to give more examples in #8078 but maybe it's not clear enough.

I feel like clearing up the docs is better than just making `+=` produce an error though, since it's not like it literally produces nonsense. It just adds a value to the end. The important part is understanding that `wildmode` is order dependent (it's kind of impossible to properly configure this option without understanding that).

Reply:
The `wildmode` option is inherently order-sensitive. Each comma-separated fragment determines the sequence in which `wildchar` behaves. Within each fragment, the colon-separated parts define simultaneous effects for `wildchar`. This is not clearly explained in the documentation. I considered rewriting the docs but chose not to include it in this PR.  

Additionally, `set wildmode=full,noselect` does not behave as expected. Ideally, the first `wildchar` should open the menu, selecting and inserting the first item (`full`), while the second `wildchar` should unselect the item and revert the command-line change (`noselect`). However, this use case is unlikely to be meaningful. In practice, the implementation simply ignores `noselect` unless it appears before the first comma.

Reply:
https://github.com/vim/vim/pull/17100